pkg/openstack/loadbalancer_test.go

```diff
@@ -1217,3 +1216,89 @@ func Test_getStringFromServiceAnnotation(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_getMemberSubnetID(t *testing.T) {
+	lbaasOpts := LoadBalancerOpts{
+		LBClasses: map[string]*LBClass{
+			"lbclassKey": {
+				MemberSubnetID: "lb-class-member-subnet-id-5678",
+			},
+		},
+		MemberSubnetID: "default-memberSubnetId",
+	}
+
+	tests := []struct {
+		name    string
+		opts    LoadBalancerOpts
+		service *corev1.Service
+		want    string
+		wantErr string
+	}{
+		{
+			name: "get member subnet id from service annotation",
+			opts: LoadBalancerOpts{},
+			service: &corev1.Service{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						ServiceAnnotationLoadBalancerMemberSubnetID: "member-subnet-id",
+						ServiceAnnotationLoadBalancerClass:          "svc-annotation-loadbalance-class",
+					},
+				},
+			},
+			want:    "member-subnet-id",
+			wantErr: "",
+		},
+		{
+			name: "get member subnet id from config class",
+			opts: lbaasOpts,
+			service: &corev1.Service{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						ServiceAnnotationLoadBalancerClass: "lbclassKey",
+					},
+				},
+			},
+			want:    "lb-class-member-subnet-id-5678",
+			wantErr: "",
+		},
+		{
+			name:    "get member subnet id from default config",
+			opts:    lbaasOpts,
+			service: &corev1.Service{},
+			want:    "default-memberSubnetId",
+			wantErr: "",
+		},
+		{
+			name: "error when loadbalancer class not found",
+			opts: LoadBalancerOpts{},
+			service: &corev1.Service{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						ServiceAnnotationLoadBalancerClass: "invalid-lb-class",
+					},
+				},
+			},
+			want:    "",
+			wantErr: "invalid loadbalancer class \"invalid-lb-class\"",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lbaas := LbaasV2{
+				LoadBalancer: LoadBalancer{
+					opts: tt.opts,
+				},
+			}
+
+			got, err := lbaas.getMemberSubnetID(tt.service)
+			if err != nil {
+				assert.EqualError(t, err, tt.wantErr)
+			} else {
+				if assert.NoError(t, err) {
+					assert.Equal(t, tt.want, got)
+				}
+			}
```

Comment:
Take a step back and look at what happens here. Your assertions are inside an `if` block. The `if` block determines which assertions will be run based on the content of **`err`**. Sometimes you need to have different assertions based on the expected values... but based on the _actual_ values? That looks suspicious.

Now look closer at what happens line by line, while you keep the variable values in your head. Imagine you expect `(tt.want == "", tt.wantErr == io.EOF)`, and your function returns `(got == "", err == nil)`. What would happen in this case?

Reply:
Don’t mind that part 
I will change it, I don’t know if to tag the PR a draft or suggestion to modify the function 
I know it can’t be merged yet because i will need to rebase and restructure it if the modification

Reply:
I made the push before your correction in that other PR and I was waiting for a review to know if the function modification looks good 
I’d have to close the PR if the removal of the `serviceConfig` isn’t authorised since there is an existing test already for the function and complete the entire modification when it is authorised 

so I apologize for the repetition, it wasn’t intentional as I have learnt the best usecase already from you and will not intentionally repeat same mistake twice

Reply:
A good workflow may be:
1. Submit a PR
2. Make the required edits until you get a LGTM
3. Rebase
4. you'll get a new LGTM
5. wait for approval
6. merge

Conflicts are extraordinarily frequent these days, because several people are submitting PRs on the same test files. That's not a normal situation, and we have to bear with that for now.

Reply:
In general, you can defer rebasing until the very last moment.

Reply:
You don't have to turn your PR into WIP, unless you realise that it is NOT ready for review and you want to avoid wasting the reviewer's time. This is NOT the case here: your PR is tidy and definitely ready for reviews.

I have asked you to turn another PR of yours to WIP, just because you mentioned that it is not ready for review.